netty/src/main/java/io/grpc/transport/netty/NettyClientTransport.java

```diff
@@ -170,33 +167,49 @@ public void start(Listener transportListener) {
      */
     b.handler(negotiationHandler);
     // Start the connection operation to the server.
-    channel = b.connect(address).channel();
+    channel = b.connect(address).addListener(new ChannelFutureListener() {
+      @Override
+      public void operationComplete(ChannelFuture future) throws Exception {
+        if (!future.isSuccess()) {
+          ChannelHandlerContext ctx = channel.pipeline().context(handler);
+          if (ctx != null) {
+            // NettyClientHandler doesn't propagate exceptions, but the negotiator will need the
+            // exception to fail any writes. Note that this fires after handler, because it is as if
+            // handler was propagating the notification.
+            ctx.fireExceptionCaught(future.cause());
+          }
+          channel.pipeline().fireExceptionCaught(future.cause());
+        }
+      }
+    }).channel();
     // Start the write queue as soon as the channel is constructed
     handler.startWriteQueue(channel);
     // Handle transport shutdown when the channel is closed.
     channel.closeFuture().addListener(new ChannelFutureListener() {
       @Override
       public void operationComplete(ChannelFuture future) throws Exception {
-        notifyTerminated(handler.connectionError());
+        Status status = handler.connectionStatus();
+        if (status == null) {
+          // We really only expect this to happen if shutdown() was called, but in that case this
+          // status is ignored.
+          status = Status.INTERNAL.withDescription("Connection closed with unknown cause");
```

Comment:
It could, but I'm not sure how much point to it there is, since either the value passed to `shutdown()` is `connectionStatus` or `connectionStatus` will be ignored.